controllers/storagecluster/exporter.go

```diff
@@ -209,9 +265,515 @@ func CreateOrUpdateServiceMonitor(r *StorageClusterReconciler, instance *ocsv1.S
+			}
+		}
+		destContainer0.Image = srcContainer0.Image
+		destContainer0.Name = srcContainer0.Name
+		destContainer0.Ports = srcContainer0.Ports
+		destContainer0.SecurityContext = srcContainer0.SecurityContext
+		destContainer0.VolumeMounts = srcContainer0.VolumeMounts
+
+		destSpec.Template.Spec.ServiceAccountName = srcSpec.Template.Spec.ServiceAccountName
+		destSpec.Template.Spec.Volumes = srcSpec.Template.Spec.Volumes
+		return nil
+	}); err != nil {
+		r.Log.Error(err, "failed to create/update metrics exporter deployment", "NamespacedName", namespacedName)
+		return err
+	}
+	return nil
+}
+
+func createMetricsExporterServiceAccount(ctx context.Context, r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	expectedServiceAccount := corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      metricsExporterName,
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/component": metricsExporterName,
+				"app.kubernetes.io/name":      metricsExporterName,
+				"app.kubernetes.io/version":   "0.0.1",
+			},
+		},
+	}
+	currentServiceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      expectedServiceAccount.Name,
+			Namespace: expectedServiceAccount.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, currentServiceAccount, func() error {
+		if currentServiceAccount.Labels == nil {
+			currentServiceAccount.Labels = make(map[string]string)
+		}
+		for lKey, lValue := range expectedServiceAccount.Labels {
+			currentServiceAccount.Labels[lKey] = lValue
+		}
+		return nil
+	})
+	return err
+}
+
+// updateMetricsExporterClusterRoleBindings function updates the
+// cluster level rolebindings for the metrics exporter in this namespace
+func updateMetricsExporterClusterRoleBindings(ctx context.Context,
+	r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	currentCRB := &rbacv1.ClusterRoleBinding{ObjectMeta: metav1.ObjectMeta{
+		Name: metricsExporterName,
+	}}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, currentCRB, func() error {
+		currentCRB.RoleRef.Name = metricsExporterName
+		currentCRB.RoleRef.Kind = "ClusterRole"
+		currentCRB.RoleRef.APIGroup = "rbac.authorization.k8s.io"
+		subjectNotFound := true
+		expectedSubject := rbacv1.Subject{
+			Kind:      "ServiceAccount",
+			Name:      metricsExporterName,
+			Namespace: instance.Namespace,
+		}
+		for _, subject := range currentCRB.Subjects {
+			if subject.Kind == expectedSubject.Kind &&
+				subject.Name == expectedSubject.Name &&
+				subject.Namespace == expectedSubject.Namespace {
+				subjectNotFound = false
+				break
+			}
+		}
+		if subjectNotFound {
+			currentCRB.Subjects = append(currentCRB.Subjects, expectedSubject)
+		}
+		return nil
+	})
+	return err
+}
+
+func createMetricsExporterSecurityContextConstraints(ctx context.Context, r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
```

Comment:
We don't want custom SCC. We should use default SCC.

Reply:
Done